Source/JavaScriptCore/builtins/JSIteratorHelperPrototype.js

```diff
@@ -51,7 +51,10 @@ function return()
     var state = @getGeneratorInternalField(generator, @generatorFieldState);
     if (state === @GeneratorStateInit) {
         @putGeneratorInternalField(generator, @generatorFieldState, @GeneratorStateCompleted);
-        @iteratorGenericClose(@getIteratorHelperInternalField(this, @iteratorHelperFieldUnderlyingIterator));
+
+        var underlyingIterator = @getIteratorHelperInternalField(this, @iteratorHelperFieldUnderlyingIterator);
+        if (!@isUndefinedOrNull(underlyingIterator))
```

Comment:
Let's make the check a bit more precise:

```suggestion
        if (underlyingIterator !== null)
```